src/backend/core/models.py

```diff
@@ -353,7 +354,7 @@ def get_queryset(self):
 
 
 # pylint: disable=too-many-public-methods
-class Document(MP_Node, BaseModel):
+class Document(DirtyFieldsMixin, MP_Node, BaseModel):
```

Comment:
Don't you want to define the `FIELDS_TO_CHECK` attribute?

https://django-dirtyfields.readthedocs.io/en/stable/customisation.html#checking-a-limited-set-of-model-fields